### PR TITLE
Allow component resources to inherit `providers` from component resources

### LINKED ISSUE #10640: Component children of component resources don't inherit their parents providers
### What happened?

I defined a VPC inside a Component resource, and parented the VPC to the wrapping component.

The resource provider has a defined AWS account and allowed account id.

The VPC was created in the account I had set in my aws environment, not in the account I expected with the explicit resource provider.

### Steps to reproduce

Create a Component Resource like so:

```typescript
import * as pulumi from "@pulumi/pulumi";
import * as aws from "@pulumi/aws";
import * as awsx from "@pulumi/awsx";
import * as classic from "@pulumi/awsx/classic";

export interface MyVpcArgs {
  cidrBlock: string;
}

export class MyVpc extends pulumi.ComponentResource {
  vpc: awsx.ec2.Vpc;
  //classicVpc: classic.ec2.Vpc;

  constructor(
    name: string,
    args: MyVpcArgs,
    opts?: pulumi.ComponentResourceOptions
  ) {
    super("jaxxstorm:index:myvpc", name, {}, opts);

    this.vpc = new awsx.ec2.Vpc(
      name,
      {
        cidrBlock: args.cidrBlock,
        subnetSpecs: [
          {
            type: "Public",
            cidrMask: 26,
          },
          {
            type: "Isolated",
            cidrMask: 26,
          },
          {
            type: "Private",
            cidrMask: 24,
          },
        ],
        natGateways: {
          strategy: "OnePerAz",
        },
        tags: {
          Name: name,
        },
      },
      { parent: this } // note the parent here
    );
  }
}
```

Then instantiate it via a provider:

```typescript
const provider = new aws.Provider("shared", {
    profile: "personal-shared_servces",
    allowedAccountIds: ["587571862190"],
})

const vpc = new myvpc.MyVpc("example", {
    cidrBlock: "172.20.0.0/20",
}, { provider: provider })
```

Note the `allowedAccountIds` and the profile I've selected.

Now, ensure your AWS credentials locally do _not_ match the profile:

```
aws sts get-caller-identity
{
    "UserId": "AROAYHKMQGUTOXWO3QUHN:lee@brig.gs",
    "Account": "565485516070",
    "Arn": "arn:aws:sts::565485516070:assumed-role/AWSReservedSSO_AWSAdministratorAccess_ee5a9f50a5aa44de/lee@brig.gs"
}
```

Now provision your Pulumi program, see where the resources get created

### Expected Behavior

The resources should be provisioned in the account defined by the resource provider

### Actual Behavior

The resources are not provisioned in the correct account

### Versions used

```
CLI          
Version      3.36.0
Go Version   go1.18.4
Go Compiler  gc

Plugins
NAME    VERSION
aws     5.10.0
awsx    1.0.0-beta.9
docker  3.2.0
nodejs  unknown

Host     
OS       darwin
Version  12.4
Arch     arm64

This project is written in nodejs: executable='/opt/homebrew/bin/node' version='v18.6.0'

Current Stack: dev

TYPE                                                 URN
pulumi:pulumi:Stack                                  urn:pulumi:dev::awsx_repro::pulumi:pulumi:Stack::awsx_repro-dev
jaxxstorm:index:myvpc                                urn:pulumi:dev::awsx_repro::jaxxstorm:index:myvpc::example
pulumi:providers:aws                                 urn:pulumi:dev::awsx_repro::pulumi:providers:aws::shared
pulumi:providers:awsx                                urn:pulumi:dev::awsx_repro::pulumi:providers:awsx::default_1_0_0_beta_9
awsx:ec2:Vpc                                         urn:pulumi:dev::awsx_repro::jaxxstorm:index:myvpc$awsx:ec2:Vpc::example
pulumi:providers:aws                                 urn:pulumi:dev::awsx_repro::pulumi:providers:aws::default_5_4_0
aws:ec2/vpc:Vpc                                      urn:pulumi:dev::awsx_repro::jaxxstorm:index:myvpc$awsx:ec2:Vpc$aws:ec2/vpc:Vpc::example
aws:ec2/subnet:Subnet                                urn:pulumi:dev::awsx_repro::jaxxstorm:index:myvpc$awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet::example-private-1
aws:ec2/subnet:Subnet                                urn:pulumi:dev::awsx_repro::jaxxstorm:index:myvpc$awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet::example-isolated-3
aws:ec2/internetGateway:InternetGateway              urn:pulumi:dev::awsx_repro::jaxxstorm:index:myvpc$awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/internetGateway:InternetGateway::example
aws:ec2/subnet:Subnet                                urn:pulumi:dev::awsx_repro::jaxxstorm:index:myvpc$awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet::example-isolated-1
aws:ec2/subnet:Subnet                                urn:pulumi:dev::awsx_repro::jaxxstorm:index:myvpc$awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet::example-isolated-2
aws:ec2/subnet:Subnet                                urn:pulumi:dev::awsx_repro::jaxxstorm:index:myvpc$awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet::example-private-3
aws:ec2/subnet:Subnet                                urn:pulumi:dev::awsx_repro::jaxxstorm:index:myvpc$awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet::example-private-2
aws:ec2/routeTable:RouteTable                        urn:pulumi:dev::awsx_repro::jaxxstorm:index:myvpc$awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/routeTable:RouteTable::example-private-1
aws:ec2/routeTable:RouteTable                        urn:pulumi:dev::awsx_repro::jaxxstorm:index:myvpc$awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/routeTable:RouteTable::example-isolated-3
aws:ec2/routeTable:RouteTable                        urn:pulumi:dev::awsx_repro::jaxxstorm:index:myvpc$awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/routeTable:RouteTable::example-isolated-1
aws:ec2/routeTable:RouteTable                        urn:pulumi:dev::awsx_repro::jaxxstorm:index:myvpc$awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/routeTable:RouteTable::example-isolated-2
aws:ec2/routeTable:RouteTable                        urn:pulumi:dev::awsx_repro::jaxxstorm:index:myvpc$awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/routeTable:RouteTable::example-private-3
aws:ec2/routeTable:RouteTable                        urn:pulumi:dev::awsx_repro::jaxxstorm:index:myvpc$awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/routeTable:RouteTable::example-private-2
aws:ec2/routeTableAssociation:RouteTableAssociation  urn:pulumi:dev::awsx_repro::jaxxstorm:index:myvpc$awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/routeTable:RouteTable$aws:ec2/routeTableAssociation:RouteTableAssociation::example-private-1
aws:ec2/routeTableAssociation:RouteTableAssociation  urn:pulumi:dev::awsx_repro::jaxxstorm:index:myvpc$awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/routeTable:RouteTable$aws:ec2/routeTableAssociation:RouteTableAssociation::example-isolated-1
aws:ec2/routeTableAssociation:RouteTableAssociation  urn:pulumi:dev::awsx_repro::jaxxstorm:index:myvpc$awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/routeTable:RouteTable$aws:ec2/routeTableAssociation:RouteTableAssociation::example-isolated-3
aws:ec2/routeTableAssociation:RouteTableAssociation  urn:pulumi:dev::awsx_repro::jaxxstorm:index:myvpc$awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/routeTable:RouteTable$aws:ec2/routeTableAssociation:RouteTableAssociation::example-isolated-2
aws:ec2/routeTableAssociation:RouteTableAssociation  urn:pulumi:dev::awsx_repro::jaxxstorm:index:myvpc$awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/routeTable:RouteTable$aws:ec2/routeTableAssociation:RouteTableAssociation::example-private-3
aws:ec2/routeTableAssociation:RouteTableAssociation  urn:pulumi:dev::awsx_repro::jaxxstorm:index:myvpc$awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/routeTable:RouteTable$aws:ec2/routeTableAssociation:RouteTableAssociation::example-private-2
aws:ec2/subnet:Subnet                                urn:pulumi:dev::awsx_repro::jaxxstorm:index:myvpc$awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet::example-public-1
aws:ec2/subnet:Subnet                                urn:pulumi:dev::awsx_repro::jaxxstorm:index:myvpc$awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet::example-public-2
aws:ec2/subnet:Subnet                                urn:pulumi:dev::awsx_repro::jaxxstorm:index:myvpc$awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet::example-public-3
aws:ec2/routeTable:RouteTable                        urn:pulumi:dev::awsx_repro::jaxxstorm:index:myvpc$awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/routeTable:RouteTable::example-public-1
aws:ec2/eip:Eip                                      urn:pulumi:dev::awsx_repro::jaxxstorm:index:myvpc$awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/eip:Eip::example-1
aws:ec2/routeTable:RouteTable                        urn:pulumi:dev::awsx_repro::jaxxstorm:index:myvpc$awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/routeTable:RouteTable::example-public-2
aws:ec2/eip:Eip                                      urn:pulumi:dev::awsx_repro::jaxxstorm:index:myvpc$awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/eip:Eip::example-2
aws:ec2/eip:Eip                                      urn:pulumi:dev::awsx_repro::jaxxstorm:index:myvpc$awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/eip:Eip::example-3
aws:ec2/routeTable:RouteTable                        urn:pulumi:dev::awsx_repro::jaxxstorm:index:myvpc$awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/routeTable:RouteTable::example-public-3
aws:ec2/routeTableAssociation:RouteTableAssociation  urn:pulumi:dev::awsx_repro::jaxxstorm:index:myvpc$awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/routeTable:RouteTable$aws:ec2/routeTableAssociation:RouteTableAssociation::example-public-1
aws:ec2/route:Route                                  urn:pulumi:dev::awsx_repro::jaxxstorm:index:myvpc$awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/routeTable:RouteTable$aws:ec2/route:Route::example-public-1
aws:ec2/routeTableAssociation:RouteTableAssociation  urn:pulumi:dev::awsx_repro::jaxxstorm:index:myvpc$awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/routeTable:RouteTable$aws:ec2/routeTableAssociation:RouteTableAssociation::example-public-2
aws:ec2/route:Route                                  urn:pulumi:dev::awsx_repro::jaxxstorm:index:myvpc$awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/routeTable:RouteTable$aws:ec2/route:Route::example-public-2
aws:ec2/natGateway:NatGateway                        urn:pulumi:dev::awsx_repro::jaxxstorm:index:myvpc$awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/natGateway:NatGateway::example-1
aws:ec2/routeTableAssociation:RouteTableAssociation  urn:pulumi:dev::awsx_repro::jaxxstorm:index:myvpc$awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/routeTable:RouteTable$aws:ec2/routeTableAssociation:RouteTableAssociation::example-public-3
aws:ec2/natGateway:NatGateway                        urn:pulumi:dev::awsx_repro::jaxxstorm:index:myvpc$awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/natGateway:NatGateway::example-3
aws:ec2/route:Route                                  urn:pulumi:dev::awsx_repro::jaxxstorm:index:myvpc$awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/routeTable:RouteTable$aws:ec2/route:Route::example-private-3
aws:ec2/route:Route                                  urn:pulumi:dev::awsx_repro::jaxxstorm:index:myvpc$awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/routeTable:RouteTable$aws:ec2/route:Route::example-public-3
aws:ec2/route:Route                                  urn:pulumi:dev::awsx_repro::jaxxstorm:index:myvpc$awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/routeTable:RouteTable$aws:ec2/route:Route::example-private-1
aws:ec2/natGateway:NatGateway                        urn:pulumi:dev::awsx_repro::jaxxstorm:index:myvpc$awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/natGateway:NatGateway::example-2
aws:ec2/route:Route                                  urn:pulumi:dev::awsx_repro::jaxxstorm:index:myvpc$awsx:ec2:Vpc$aws:ec2/vpc:Vpc$aws:ec2/subnet:Subnet$aws:ec2/routeTable:RouteTable$aws:ec2/route:Route::example-private-2
pulumi:providers:pulumi                              urn:pulumi:dev::awsx_repro::pulumi:providers:pulumi::default


Found no pending operations associated with dev

Backend        
Name           pulumi.com
URL            https://app.pulumi.com/jaxxstorm
User           jaxxstorm
Organizations  jaxxstorm, lbrlabs, team-ce, demo, pulumi

Dependencies:
NAME            VERSION
@pulumi/aws     5.10.0
@pulumi/awsx    1.0.0-beta.9
@pulumi/pulumi  3.36.0
@types/node     14.18.22

Pulumi locates its logs in /var/folders/rk/2c715ngd14jc4c7d3659m7pm0000gn/T/ by default

```

### Additional context

It is entirely possible I am misunderstanding something here

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

### PR DESCRIPTION
**Pull request by iwahbe**
# Description

Fixes #10640 

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works

- [x] I have run `make changelog` and committed the `changelog/pending/<file>` documenting my change

- [ ] Yes, there are changes in this PR that warrants bumping the Pulumi Service API version

### CODE REVIEW

**Review by Frassle — COMMENTED** [review]
Can't this be fixed in the engine rather than the SDKs?

**Comment by iwahbe** [comment]
@Frassle This can be fixed in the engine. It's a cleaner fix but requires bifurcating `providers` logic. It needs to occur before remote component resources are constructed, which means it needs to live in the `resmon`.

**Comment by justinvp** [comment]
Any tests?

**Comment by iwahbe** [comment]
> Any tests?

I was trying to come up with a way to test this without writing a full integration test. I'll add an integration test shortly.

**Comment by Frassle** [comment]
You could test this in pulumi_tests.go, there's a lot of other engine tests there.

File: sdk/python/lib/pulumi/runtime/resource.py [hunk 1]
```diff
@@ -154,7 +154,7 @@ async def prepare_resource(
     # For remote resources, merge any provider opts into a single dict, and then create a new dict with all of the
     # resolved provider refs.
     provider_refs: Dict[str, Optional[str]] = {}
-    if remote and opts is not None:
+    if (remote or not custom) and opts is not None:
```
Comment (Frassle) [inline sdk/python/lib/pulumi/runtime/resource.py:157]: This is just "not custom" now isn't it? Resources are either LocalComponent, RemoteComponent, or Custom (remote=true, custom=true should be inexpressible). Also is a similar change not needed in the dotnet code?

**Comment by iwahbe** [comment]
Writing the necessary test is taking longer than I thought. This is what I'm thinking:
We need a test that looks like this:
1. Create provider `p` of package `pkgA`.
2. Create language specific component `c1`, passing providers=[p].
3. `c1` creates MLC component `c2` of package `pkgA` as a child, passing `parent=c1` on creation.
4. `c2` creates custom resource `r` of package `pkgB`.
5. We assert that `r` was created with provider `p`.

File: sdk/python/lib/pulumi/runtime/resource.py (hunk 1 above)
Comment (iwahbe) [reply sdk/python/lib/pulumi/runtime/resource.py:157]: I couldn't find the equivalent logic. It is just a branch on `custom`: https://github.com/pulumi/pulumi/blob/85cfd79d32647019c5edb6ff45130aff9c66d10a/sdk/dotnet/Pulumi/Resources/Resource.cs#L221-L259

**Comment by iwahbe** [comment]
> You could test this in pulumi_tests.go, there's a lot of other engine tests there.

It doesn't look like `pulumi_test.go` puts `resmon` under test (which is where the change in engine logic is located).

**Comment by Frassle** [comment]
> It doesn't look like pulumi_test.go puts resmon under test (which is where the change in engine logic is located).

"pkg/engine/lifecycletest/pulumi_test.go"? That tests resmon because it tests via sending grpc messages to a resource monitor (either fully over grpc, or just method calls).

**Comment by iwahbe** [comment]
> > It doesn't look like pulumi_test.go puts resmon under test (which is where the change in engine logic is located).
> 
> "pkg/engine/lifecycletest/pulumi_test.go"? That tests resmon because it tests via sending grpc messages to a resource monitor (either fully over grpc, or just method calls).

Having though about it longer, I think it's still better to have an integration test here. We ensure that the SDK and the engine are behaving appropriately together.

File: tests/integration/construct_component/testcomponent-go/main.go [hunk 2]
```diff
@@ -41,7 +41,7 @@ func NewResource(ctx *pulumi.Context, name string, echo pulumi.Input,
 	opts ...pulumi.ResourceOption) (*Resource, error) {
 	args := &ResourceArgs{Echo: echo}
 	var resource Resource
-	err := ctx.RegisterResource("testcomponent:index:Resource", name, args, &resource, opts...)
+	err := ctx.RegisterResource(providerName+":index:Resource", name, args, &resource, opts...)
```
Comment (Frassle) [inline tests/integration/construct_component/testcomponent-go/main.go:44]: I don't see providerName declared in here?

File: pkg/resource/deploy/source_eval.go [hunk 3]
```diff
@@ -1115,6 +1131,13 @@ func (rm *resmon) RegisterResource(ctx context.Context,
 
 	// If this is a remote component, fetch its provider and issue the construct call. Otherwise, register the resource.
 	var result *RegisterResult
+	if !custom {
+		defer func() {
```
Comment (Frassle) [inline pkg/resource/deploy/source_eval.go:1135]: Why is this defered? Can't we always just store the provider list as it comes in?

File: tests/integration/construct_component/testcomponent-go/main.go (hunk 2 above)
Comment (iwahbe) [reply tests/integration/construct_component/testcomponent-go/main.go:44]: Its declared on line 104. 
https://github.com/pulumi/pulumi/blob/12c75e91dbe3b9b8ea0ce0e043d6fbfdc9a19a06/tests/integration/construct_component/testcomponent-go/main.go#L104

File: pkg/resource/deploy/source_eval.go (hunk 3 above)
Comment (iwahbe) [reply pkg/resource/deploy/source_eval.go:1135]: We know the value when we declare the deferred function (L1135), but we don't know the key (URN) to store it at. Note that the defer only finishes the store if we successfully stored an URN.

File: pkg/resource/deploy/source_eval.go [hunk 4]
```diff
@@ -959,6 +961,20 @@ func (rm *resmon) RegisterResource(ctx context.Context,
 		t = tokens.Type(req.GetType())
 	}
 
+	// We handle update the provides map to include the providers field of the parent if
```
Comment (justinvp) [inline pkg/resource/deploy/source_eval.go:964]: Nit:

```suggestion
	// We handle updating the providers map to include the providers field of the parent if
```

? or was something else intended

File: pkg/resource/deploy/source_eval.go (hunk 3 above)
Comment (justinvp) [reply pkg/resource/deploy/source_eval.go:1135]: Should we just move this code down to L1197 when we have the result, so we don't have to have the defer here?

If there's a reason it needs to live here with the defer, it needs a comment explaining why.

File: tests/integration/component_setup.sh [hunk 5]
```diff
@@ -39,11 +39,22 @@ setup_nodejs() (
   fi
 )
 
+# cd testcomponent-go && go build -o out && \
+#   pulumi plugin install resource testcomponent v0.1.0 --reinstall -f out && \
+#   cd .. && cd testcomponent2-go && go build -o out && \
+#   pulumi plugin install resource secondtestcomponent v0.1.0 --reinstall -f out && \
+#   cd .. && cd go && pulumi up || cd ..
```
Comment (justinvp) [inline tests/integration/component_setup.sh:46]: What are these comments for? Do they need to be kept?

File: tests/integration/construct_component/testcomponent2-go/main.go [hunk 6]
```diff
@@ -0,0 +1,274 @@
+// Copyright 2016-2021, Pulumi Corporation.  All rights reserved.
```
Comment (justinvp) [inline tests/integration/construct_component/testcomponent2-go/main.go:1]: Nit:

```suggestion
// Copyright 2016-2022, Pulumi Corporation.  All rights reserved.
```

File: tests/integration/construct_nested_component/go/Pulumi.yaml [hunk 7]
```diff
@@ -0,0 +1,3 @@
+name: construct_component_go
```
Comment (justinvp) [inline tests/integration/construct_nested_component/go/Pulumi.yaml:1]: Nit:
 
```suggestion
name: construct_nested_component_go
```

File: tests/integration/construct_nested_component/go/main.go [hunk 8]
```diff
@@ -0,0 +1,120 @@
+// Copyright 2016-2020, Pulumi Corporation.  All rights reserved.
```
Comment (justinvp) [inline tests/integration/construct_nested_component/go/main.go:1]: Nit:
```suggestion
// Copyright 2016-2022, Pulumi Corporation.  All rights reserved.
```

**Review by justinvp — APPROVED** [review]
LGTM. It took me a while to parse what the test was doing (maybe my brain is just foggy from being sick).

Isn't this only testing the Go SDK? Any reason we're not testing the other SDKs?

Does Java have the same problem? Should we be sending the providers to the engine there as well?

File: tests/integration/component_setup.sh (hunk 5 above)
Comment (iwahbe) [reply tests/integration/component_setup.sh:46]: The comment is the one liner necessary to run the test manually. They can be removed.

**Comment by iwahbe** [comment]
Java has the same semantics as .NET, which already did the correct thing. I added a typescript test here. It was only Go because one SDK was sufficient to test that the engine did there right thing.

**Pull request merged**